### ISSUE TITLE
Can libheif support cmake version 3.10.2 (Ubuntu 18)?

### ISSUE THREAD

**Issue by urashimamei** [strukturag/libheif#1325]
Hi,

I have a server that is Ubuntu 18 and cannot upgrade, because many people is using it and some work needs to run in this version.
Unfortunately, I have to make the server support iOS 18 photos at the same time.
How to make the build support CMake 3.10.2?
Any advise is fine. Thanks:))

**Comment by farindk** [comment]
Alternatively to using an old cmake, you could try to cherry-pick the commits 4d76bae and 3a6f8cd. See #1190.

**Comment by bobfriesenhahn** [comment]
When I encountered this same issue, I built cmake 3.22.1 from original sources.  It seems that a special installation prefix or cmake executable name could be used to differentiate from the older version.

**Comment by urashimamei** [comment]
> Alternatively to using an old cmake, you could try to cherry-pick the commits [4d76bae](https://github.com/strukturag/libheif/commit/4d76bae4656caa38d01320a96a730b38bb4b4cde) and [3a6f8cd](https://github.com/strukturag/libheif/commit/3a6f8cdec345e467a538f7c728376fa2e5f3031c). See #1190.

Do you mean using the source 1.16.2 (this is the last version support Ubuntu 18) and cherry-pick the commits you suggest for the feature? I will give it a try and update the result here. Thank you so much.

**Comment by urashimamei** [comment]
> When I encountered this same issue, I built cmake 3.22.1 from original sources. It seems that a special installation prefix or cmake executable name could be used to differentiate from the older version.

@bobfriesenhahn I have the same thought but that would make the system dirty... This will be my last option if any other method not working. Thank you for the kind advice!!

**Comment by urashimamei** [comment]
I've downloaded the executable CMake version 3.28 to compile the latest libheif source. Thank you all for your help. 

I tried using an older version and cherry-pick the commits ＠farindk suggested but still encountered issues, so I ultimately resolved it this way. Since the server is shared among everyone and others can use this project, I didn't want to use a non-installed version of the new CMake. Anyway, this problem has been solved. Thank you to everyone who offered suggestions.
